lib/application.js

```diff
@@ -158,6 +158,11 @@ module.exports = class Application extends Emitter {
       keys: this.keys,
       secure: request.secure
     });
+    if (req && req.headers) {
+      request.ip = request.ips[0] || req.socket.remoteAddress || '';
+    } else {
+      request.ip = ''; // We are in a badly written test :P
```

Comment:
That's the current behavior, changing that would be breaking for no reason. Also, it's better because it keeps the type.